src/libraries/Common/src/System/Security/Cryptography/ECDsaOpenSsl.cs

```diff
@@ -242,16 +273,18 @@ public override bool VerifyHash(byte[] hash, byte[] signature)
                     signatureFormat.ToString());
             }
 
-            SafeEcKeyHandle key = _key.Value;
-            int verifyResult = Interop.Crypto.EcDsaVerify(hash, toVerify, key);
-            return verifyResult == 1;
+            using (SafeEvpPKeyCtxHandle ctx = CreateCtx())
+            {
+                ctx.ConfigureForECDSAVerify();
+                return ctx.VerifyHashCore(hash, toVerify);
```

Comment:
I don't think we need SafeEvpPKeyCtxHandle at all for this.

There should just be one shim function that takes key, hash, and toVerify (and lengths as needed).

Reply:
I moved this code to use SafeEvpPKeyCtxHandle mainly for two reasons:
- We can now share some code between RSA/ECDSA/ECDH - specifically passing provider handle is much easier to plumb automatically since we can hide passing in the handle as part of context creation
- I felt like we'd want to eventually get rid of the native component completely (or at least reduce it significantly) and do everything (most) in managed and this was a step in that direction - also ECDH was already using context and code felt cleaner than RSA and ECDsa

Is there any benefit of doing everything in native? You'd need to upref key (in reference to your other comment) even in native code for this to work multithreaded so that doesn't isolate that as well